Display SMW banner when needed

noname1706 opened the pull request:
[3.1.x]

ERM 19184

osnard commented:
There are two issues with this approach:

1. `class_exists` is not necessarily a good way to check is an extension is enabled.
2. Settings that belong to an extension should be set in the proper location. The "skin" should not need to know about whether or not an extension is enabled. And it should not need to check explicitly. Rather mode the whole "SMW-Footer-Icon-Registration" from `hallowelt/mediawiki:settings.d/080-BlueSpiceCalumma.php` to `BlueSpice/mediawiki:050-BlueSpiceSemanticData.php`. Maybe add some logic to maintain the sort order.

noname1706 commented:
> There are two issues with this approach:
> 
>     1. `class_exists` is not necessarily a good way to check is an extension is enabled.
> 
>     2. Settings that belong to an extension should be set in the proper location. The "skin" should not need to know about whether or not an extension is enabled. And it should not need to check explicitly. Rather mode the whole "SMW-Footer-Icon-Registration" from `hallowelt/mediawiki:settings.d/080-BlueSpiceCalumma.php` to `BlueSpice/mediawiki:050-BlueSpiceSemanticData.php`. Maybe add some logic to maintain the sort order.

1)Changed wrong approach with class_exists() function;
2)Moved SMW banner setting to 050-BlueSpiceSemanticData.php
and added unset logic there as well. But when I unset gray one - blue one appears on it's place :) So will investigate a bit further. Also created branch for it in gitlab repo  - because it has to be done there, as I understood.

Thx.

osnard commented:
Will be solved in another place
